src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -383,6 +383,48 @@ public IndexTerm next() {
     return docVector;
   }
 
+  /**
+   * Returns the term position mapping for a particular document. Note that this method explicitly returns
+   * {@code null} if the document does not exist (as opposed to an empty map), so that the caller is explicitly forced
+   * to handle this case.
+   *
+   * @param reader index reader
+   * @param docid collection docid
+   * @return term position mapping for a particular document or {@code null} if document does not exist.
+   * @throws IOException if error encountered during query
+   * @throws NotStoredException if the term vector is not stored
+   */
+  public static Map<String, List<Long>> getTermPositions(IndexReader reader, String docid) throws IOException, NotStoredException {
+    int ldocid = convertDocidToLuceneDocid(reader, docid);
+    if (ldocid == -1) {
+      return null;
+    }
+    Terms terms = reader.getTermVector(ldocid, IndexArgs.CONTENTS);
+    if (terms == null) {
+      throw new NotStoredException("Document vector not stored!");
+    }
+    TermsEnum termIter = terms.iterator();
+    if (termIter == null) {
+      throw new NotStoredException("Document vector not stored!");
+    }
+
+    Map<String, List<Long>> termPosition = new HashMap<>();
```

Comment:
Just noticed this - do we need to use long? I don't think documents are going to be that looong? How about just `List<Integer>`?

Reply:
Yes. `position_iter.nextPosition()` return int. Using long is unnecessary.  I have changed the declaration of `termPostion` accordingly.
I also discovered that `Long termFreq = termIter.totalTermFreq();` introduce autoboxing. I have changed `Long` to `long` to avoid this.

Reply:
yes, autoboxing is inefficient, but since you have `List<Integer>`, you'll have to convert it back into an object anyway...